torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -245,7 +245,24 @@ void PropagateShapeOnNode(Node * node, bool insert_expands) {
       }
       return;
     }
-    case prim::TensorToNum:
+    case prim::ScalarToNum:
+      if (auto err = checkScalarToNum(node->input()->type(), node->output()->type())) {
+        throw std::runtime_error(*err);
+      }
+    case prim::TensorToNum: {
+      if (auto ten = node->input()->type()->cast<TensorType>()) {
+        int64_t numel = 1;
+        for (int64_t size : ten->sizes()) {
+          numel *= size;
+        }
+        if (numel != 1) {
+          std::stringstream ss;
+          ss << "Tensor with " << numel << " elements could not be converted to Scalar\n";
+          throw std::runtime_error(ss.str());
```

Comment:
Shape analysis should never fail, even if the code when run _would_ fail. This would be as if the C++ compiler would crash in an optimization if it noticed the code it was compiling would throw an exception.  TensorToNum should just leave the number type as is.